### ISSUE TITLE
path touching tile edge permanently altered if touched by path-building "ghost tile"

### ISSUE THREAD

**Issue by cheweytoo** [OpenRCT2/OpenRCT2#18494]
### Operating System

Ubuntu Linux 20.04.5

### OpenRCT2 build

afd0346 develop, AppImage

### Base game

RollerCoaster Tycoon 2

### Describe the issue

Paths lengthened to touch the tile edge will be shortened to their "not quite to the edge" default if that edge is touched by the path building "ghost tile" - without building or clicking anything:

https://user-images.githubusercontent.com/29800630/199575794-64274cf5-9a31-4a03-b9ec-a0f61d15d0a6.mp4

### Steps to reproduce

- build path
- use tile inspector to lengthen path to touch tile edge
- touch tile edge with path-building "ghost tile"

attached park for testing

### Attachments

[ghost-shortened_paths.zip](https://github.com/OpenRCT2/OpenRCT2/files/9923290/ghost-shortened_paths.zip)

**Comment by Broxzier** [comment]
Manually adjusting the path edges is not something that you can normally do, so they reset to their default, intended way. It would be ideal if ghodt pieces only adjusted the visuals instead of the actual elements.

**Comment by cheweytoo** [comment]
Yeah. The action of _not_ building something (in other words: a "no action" action) should never alter already built things – regardless of how those got to how they are in the first place.
